### PR TITLE
Change test scenarios for rule rpm_verify_permissions

### PR DESCRIPTION
**Pull request by jan-cerny**
Test scenario fresh_system.pass.sh is removed, because it's wrong,
because it doesn't set up the system in any way, it only checks if all
the files have correct permissions according to RPM. However, on test
machines, there are usually some files which have incorrect permissions.
Also, on a default RHEL 7 installation, there are files which have
incorrect permissions.  Instead of fresh_system.pass.sh we add a
different test scenario that ensures every file has correct permissions
according to RPM after the set up phase. The scenario basically contains
the remediation for this rule.

### CODE REVIEW

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+FILES_WITH_INCORRECT_PERMS=($(rpm -Va --nofiledigest | awk '{ if (substr($0,2,1)=="M") print $NF }'))
+
+# For each file path from that list:
+# * Determine the RPM package the file path is shipped by,
+# * Include it into SETPERMS_RPM_LIST array
+
+for FILE_PATH in "${FILES_WITH_INCORRECT_PERMS[@]}"
+do
+	RPM_PACKAGE=$(rpm -qf "$FILE_PATH")
+	SETPERMS_RPM_LIST=("${SETPERMS_RPM_LIST[@]}" "$RPM_PACKAGE")
```
Comment (matejak) [inline tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: Use `SETPERMS_RPM_LIST+=("$RPM_PACKAGE")` to append to the array instead.

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+# profiles = xccdf_org.ssgproject.content_profile_pci-dss
+
+
+# Declare array to hold list of RPM packages we need to correct permissions for
+declare -a SETPERMS_RPM_LIST
+
+# Create a list of files on the system having permissions different from what
+# is expected by the RPM database
+FILES_WITH_INCORRECT_PERMS=($(rpm -Va --nofiledigest | awk '{ if (substr($0,2,1)=="M") print $NF }'))
```
Comment (matejak) [inline tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: Side note: This line assumes that there is no whitespace in filenames, so using an array here doesn't add any value. I think that it is OK though.

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# * Include it into SETPERMS_RPM_LIST array
+
+for FILE_PATH in "${FILES_WITH_INCORRECT_PERMS[@]}"
+do
+	RPM_PACKAGE=$(rpm -qf "$FILE_PATH")
+	SETPERMS_RPM_LIST=("${SETPERMS_RPM_LIST[@]}" "$RPM_PACKAGE")
+done
+
+# Remove duplicate mention of same RPM in $SETPERMS_RPM_LIST (if any)
+SETPERMS_RPM_LIST=( $(echo "${SETPERMS_RPM_LIST[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' ') )
```
Comment (matejak) [inline tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: Consider using associative arrays, they work even in RHEL6, and you don't have to do this echo stuff:

https://stackoverflow.com/a/28368319

**Comment by jan-cerny** [comment]
@matejak Should I do those changes also in https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_permissions/bash/shared.sh from where this is copy-pasted?

**Comment by jan-cerny** [comment]
```
[jcerny@thinkpad tests{fix_test_rpm_verify_permissions}]$ python3 test_suite.py rule --datastream /home/jcerny/scap-security-guide/build/ssg-rhel7-ds.xml --libvirt qemu:///system latest-rhel-7 rpm_verify_permiss
ions
Setting console output to log level INFO
INFO - The DataStream contains 2 Benchmarks
INFO - 0 - scap_org.open-scap_cref_ssg-rhel7-xccdf-1.2.xml
INFO - 1 - scap_org.open-scap_cref_ssg-rhel7-pcidss-xccdf-1.2.xml
INFO - Selected Benchmark is 0
INFO - To select a different Benchmark, use --xccdf-id-number option.
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/scap-security-guide/tests/logs/rule-custom-2019-05-16-0948/test_suite.log                                                                                                        
INFO - xccdf_org.ssgproject.content_rule_rpm_verify_permissions
INFO - Script bad_permissions.fail.sh using profile xccdf_org.ssgproject.content_profile_pci-dss OK
INFO - Script all_permissions_ok.pass.sh using profile xccdf_org.ssgproject.content_profile_pci-dss OK
[jcerny@thinkpad tests{fix_test_rpm_verify_permissions}]$
```

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh (hunk 3 above)
Comment (jan-cerny) [reply tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: done

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh (hunk 1 above)
Comment (jan-cerny) [reply tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: when using associative arrays, is this still applicable?

File: tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh (hunk 2 above)
Comment (jan-cerny) [reply tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh]: OK

**Comment by matejak** [comment]
This is great, thanks!

**Comment by jan-cerny** [comment]
@matejak What is your suggestion on my question?

**Pull request merged**